python/taichi/lang/__init__.py

```diff
@@ -289,6 +289,13 @@ def svd(A, dt=None):
     return svd(A, dt)
 
 
+def eig(A, dt=None):
+    if dt is None:
+        dt = impl.get_runtime().default_fp
+    from .linalg import eig
+    return eig(A, dt)
```

Comment:
While this style is probably following how `svd()` is implemented, it could be a bit confusing to have multiple `eig()` symbols. I'd suggest to change to this:

```py
def eig(A, dt=None):
    if dt is None:
        dt = impl.get_runtime().default_fp
    from taichi.lang import linalg
    if A.n == 2:
      return linalg.eig2x2(A, dt)
    raise Exception("Eigen solver only supports 2D matrices.")
```

Reply:
Actually, I think it's better to split into two PRs. This makes the review process easier ;-)

Reply:
Sure. I'll fix it.